sklearn/tests/test_metaestimators_metadata_routing.py

```diff
@@ -356,6 +356,15 @@ def enable_slep006():
         "cv_name": "cv",
         "cv_routing_methods": ["fit"],
     },
+    {
+        "metaestimator": SelfTrainingClassifier,
+        "estimator_name": "base_estimator",
+        "estimator": "classifier",
+        "X": X,
+        "y": y,
+        "preserves_metadata": True,
+        "estimator_routing_methods": ["fit"],
```

Comment:
we can add the other methods here, can't we?

Reply:
I can add `predict` as well. But when I add `score` and `decision_function`, I get the following error:

```
 AssertionError: Expected dict_keys(['sample_weight']) vs dict_keys([])
```

Reply:
You might need to fix the Consumer object as well for this to work. Pipeline is a different story cause it has several steps and transforming data in between and all, this object is much simpler and it might not need the separate tests at all.

Reply:
Okay, I think I got this working now. I reverted the Pipeline `SimpleEstimator` back and leveraged the entire framework in `test_metaestimators_metadata_routing.py` instead. The unit tests seem to pass locally.

Reply:
Actually I think when I add `score` to the list of methods to test, I get the following errors. It's challenging for me to figure out why this is occurring tho tbh. Was wondering if you had any pointers?

```
                # `set_{method}_request({metadata}==True)` on the underlying objects
>               set_requests(
                    estimator,
                    method_mapping=method_mapping,
                    methods=[method_name],
                    metadata_name=key,
                )

sklearn/tests/test_metaestimators_metadata_routing.py:692: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
sklearn/tests/test_metaestimators_metadata_routing.py:548: in set_requests
    set_request_for_method(**{metadata_name: value})
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

args = (), kw = {'metadata': True}

    def func(*args, **kw):
        """Updates the request for provided parameters
    
        This docstring is overwritten below.
        See REQUESTER_DOC for expected functionality
        """
        if not _routing_enabled():
            raise RuntimeError(
                "This method is only available when metadata routing is enabled."
                " You can enable it using"
                " sklearn.set_config(enable_metadata_routing=True)."
            )
    
        if self.validate_keys and (set(kw) - set(self.keys)):
>           raise TypeError(
                f"Unexpected args: {set(kw) - set(self.keys)} in {self.name}. "
                f"Accepted arguments are: {set(self.keys)}"
            )
E           TypeError: Unexpected args: {'metadata'} in score. Accepted arguments are: {'sample_weight'}

sklearn/utils/_metadata_requests.py:1264: TypeError
============================================== short test summary info ===============================================
FAILED sklearn/tests/test_metaestimators_metadata_routing.py::test_error_on_missing_requests_for_sub_estimator[SelfTrainingClassifier]
FAILED sklearn/tests/test_metaestimators_metadata_routing.py::test_setting_request_on_sub_estimator_removes_error[SelfTrainingClassifier]
```